src/custom-elements/src/components/search-input/search-input.tsx

```diff
@@ -200,15 +201,13 @@ export class SearchInput {
     dropdown.show();
   }
 
-  _handleSelectItem = (ev: Event): void => {
+  _handleSelectItem = (ev: Event, item: DropdownMenuItem): void => {
     ev.preventDefault();
 
-    const title = (ev.target as HTMLAnchorElement).innerText;
-    const href = (ev.target as HTMLAnchorElement).getAttribute('href');
     const dropdown = this._getAutocompleteDropdown();
 
-    this.selectedItem = { title, href };
-    this.value = title;
+    this.selectedItem = item;
```

Comment:
With this we come back to the previos issue, types are not strict and if we pass an item as 

```
{ 
   title: 'something',
    href: 'somehref',
    foo: 'bar'
}
```

the event.detail will have `foo`.

The fastest way to ensure that the ev.detail (and selected item) follows the type would be to to:

```suggestion
    const { title, href, id } = item ?? {};
    this.selectedItem = {title, href, id};
```

A more broad approach would be to map `this.menuItems` on component load:

```
try {
    this.menuItems = this.menuItems.map(
        ({ title, href, id }) => ({ title, href, id })
    );
} catch .... trow 'Menu items are not of type DropdownMenuItems' 
```